os_fdp_adapters/xml-adapter/xml_to_fdp.py

```diff
@@ -0,0 +1,100 @@
+import sys
+import csv
+import json
+import tempfile
+import os
+import six
+
+sys.path.append('..')
+from common.fetchers import fetch_local_filename
+from common.urls import wrap
+
+csvDelimiter = ','
+
+
+def to_bytes(x):
+    if six.PY2 and type(x) is six.text_type:
+        return x.encode('utf8')
+    return x
+
+
+def gen_csv_file(url):
+    """
+    :param url: url can be the absolute address of an xml file, or  an xml file located at web (http://..../<name>.xml
+    :return: list of csv files, each contains a data-table extracted from the input xml file
+    """
+    orginBaseName = os.path.basename(url).strip(' ')
+    if url.find('http') == -1:
+        url = os.path.abspath(url)
+    with tempfile.NamedTemporaryFile(delete=False) as fd:
+        cmd = "xml2csv {} -a 135 >> {}".format(url, fd.name)
+        os.system(cmd)
+        return split_into_single_csvs(fd.name, orginBaseName)
+
+
+def split_into_single_csvs(filename, orginBaseName, tableSeparator="\n\n\n"):
```

Comment:
This function should NOT write all the CSV files to disk.
You should have one function for getting the list of CSVs (for the `json` request).
You should have another function for extracting _one_ CSV file into stdout (without saving it to disk).
**No test specific code is allowed!**